xds/internal/client/client_xds.go

```diff
@@ -336,6 +334,9 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger)
 			if totalWeight != wcs.GetTotalWeight().GetValue() {
 				return nil, fmt.Errorf("route %+v, action %+v, weights of clusters do not add up to total total weight, got: %v, want %v", r, a, wcs.GetTotalWeight().GetValue(), totalWeight)
 			}
+			if totalWeight == 0 {
+				return nil, fmt.Errorf("route %+v, action %+v, has no cluster in WeightedCluster action", r, a)
+			}
```

Comment:
Is a zero weight for a cluster valid?

If so, then is total weight of zero valid if there are clusters?

If not, should we nack if we get a zero value on line 330?

Reply:
C and Java have different behaviors.
C would ignore weight 0, and if there is no remaining cluster, NACK.
Java processes weight 0 as normal.
Need to find what Envoy does.

Will make the fix after there's a conclusion.

Reply:
Made the change to ignore 0-weight clusters. And will still NACK if there's no clusters remaining after filtering.